pkg/simpleschema/transform.go

```diff
@@ -275,6 +275,25 @@ func (tf *transformer) applyMarkers(schema *extv1.JSONSchemaProps, markers []*Ma
 				},
 			}
 			schema.XValidations = validation
+		case MarkerTypeImmutable:
+			switch isImmutable, err := strconv.ParseBool(marker.Value); {
+			case err != nil:
+				return fmt.Errorf("failed to parse immutable marker value: %w", err)
+			case isImmutable:
+				immutableValidation := []extv1.ValidationRule{
+					{
+						Rule:    "self == oldSelf",
+						Message: "field is immutable",
+					},
+				}
+				if schema.XValidations == nil {
+					schema.XValidations = immutableValidation
+				} else {
+					schema.XValidations = append(schema.XValidations, immutableValidation...)
+				}
```

Comment:
append works on nil arrays
```suggestion
		        schema.XValidations = append(schema.XValidations, immutableValidation...)
```